src/main/java/seedu/address/model/ModelManager.java

```diff
@@ -241,4 +255,12 @@ public void setDeadlineTaskFilter(Predicate<? super DeadlineTask> predicate) {
         filteredDeadlineTasks.setPredicate(predicate);
     }
 
+    @Override
+    public void refilterDeadlineTaskFilteredList() {
+        Predicate<? super DeadlineTask> predicate = filteredDeadlineTasks.getPredicate();
+        assert predicate != null;
+        setDeadlineTaskFilter(null);
+        setDeadlineTaskFilter(predicate);
```

Comment:
Ah.. no wonder.. If I'm guessing correctly it's because the UI is not updating? (Because you did not obey the "Tasks are immutable" rule)